viewer/src/cache/RequestCache.ts

```diff
@@ -2,10 +2,9 @@
  * Copyright 2020 Cognite AS
  */
 
-export type RequestDelegate<T_ID, T> = (id: T_ID) => Promise<T>;
+export type RequestDelegate<Key, Data, Result> = (id: Key, data: Data) => Result;
```

Comment:
You also need to pass in the actual data, which in the case for `FetchedSector => ParsedSector` would be the fetched data, or in the case of `ParsedSector => ConsumedSector` would be the parsed data. We cannot compare the actual data to do a true memoization, since that would be very costly (comparing megabytes of array buffers).